pontoon/sync/tests/test_changeset.py

```diff
@@ -300,3 +301,41 @@ def test_obsolete_db(self):
         self.changeset.execute()
         self.main_db_entity.refresh_from_db()
         assert_true(self.main_db_entity.obsolete)
+
+    def test_multiple_locale_authors(self):
+        """
+        Test if authors are allocated per locale properly.
+        Regression tests against bug 1214722.
+
+        Contributors should map to locales following way:
+            first_locale -> first_user, third_user
+            second_locale -> second_user
+        """
+        first_entity, second_entity = EntityFactory.create_batch(2)
+        first_locale, second_locale = LocaleFactory.create_batch(2)
+        first_author, second_author, third_author = UserFactory.create_batch(3)
+
+        TranslationFactory.create(locale=first_locale, user=first_author, entity=first_entity)
+        TranslationFactory.create(locale=second_locale, user=second_author, entity=first_entity)
+        TranslationFactory.create(locale=first_locale, user=third_author, entity=first_entity)
+        # Because of logic contained in Translation.save() we have too update
+        # approved field here.
+        Translation.objects.update(approved=True)
+
```

Comment:
If I understand correctly, you translated first_entity into first_locale with first_author first and later with third_author? And made both of these translations approved? That's not something we want, there can only be 1 approved translation for each string. And if we have multiple translations for the same string in the same locale, we should only attribute it to the user that committed the approved translation in VCS.